model/collection.py

```diff
@@ -794,9 +794,7 @@ def delete(self, search_index=None):
                 _db.commit()
             if work and search_index and not work.license_pools:
                 search_index.remove_work(work)
-                # TODO: ideally we'd delete the Work itself, but
-                # that's another thing we don't normally do, and it
-                # might have side effects we haven't considered.
+                _db.delete(work)
```

Comment:
Since the WorkReaper does exactly this -- deletes Works from the database because they have no license pools -- I no longer see a problem with deleting Works immediately in the CirculationReaper as they lose their license pools.